bundles/org.openhab.binding.vesync/src/main/java/org/openhab/binding/vesync/internal/handlers/VeSyncBaseDeviceHandler.java

```diff
@@ -0,0 +1,479 @@
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link VeSyncBaseDeviceHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author David Goodyear - Initial contribution
+ */
+@NonNullByDefault
+public abstract class VeSyncBaseDeviceHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(VeSyncBaseDeviceHandler.class);
+
+    private static final String MARKER_INVALID_DEVICE_KEY = "---INVALID---";
+
+    private static final int CACHE_TIMEOUT_SECOND = 5;
+
+    private int activePollRate = -2; // -1 is used to deactivate the poll, so default to a different value
+
+    private @Nullable ScheduledFuture<?> backgroundPollingScheduler;
+    private final Object pollConfigLock = new Object();
+
+    protected @Nullable Channel findChannelById(final String channelGroupId) {
+        // return getThing().getChannels().stream().anyMatch(x -> x.getUID().getId().equals(channelGroupId));
+        return getThing().getChannel(channelGroupId);
+    }
+
+    protected ExpiringCache<String> lastPollResultCache = new ExpiringCache<>(Duration.ofSeconds(CACHE_TIMEOUT_SECOND),
+            VeSyncBaseDeviceHandler::expireCacheContents);
+
+    private static @Nullable String expireCacheContents() {
+        return null;
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        super.channelLinked(channelUID);
+
+        scheduler.execute(this::pollForUpdate);
+    }
+
+    protected void setBackgroundPollInterval(final int seconds) {
+        if (activePollRate == seconds) {
+            return;
+        }
+        logger.debug("Reconfiguring devices background polling to {} seconds", seconds);
+
+        synchronized (pollConfigLock) {
+            final ScheduledFuture<?> job = backgroundPollingScheduler;
+
+            // Cancel the current scan's and re-schedule as required
+            if (job != null && !job.isCancelled()) {
+                job.cancel(true);
+                backgroundPollingScheduler = null;
+            }
+            if (seconds > 0) {
+                logger.trace("Device data is polling every {} seconds", seconds);
+                backgroundPollingScheduler = scheduler.scheduleWithFixedDelay(this::pollForUpdate, seconds, seconds,
+                        TimeUnit.SECONDS);
+            }
+            activePollRate = seconds;
+        }
+    }
+
+    @NotNull
+    protected String deviceLookupKey = MARKER_INVALID_DEVICE_KEY;
+
+    public void configurationUpdated(Thing thing) {
+        logger.debug("DETECTED CONFIG UPDATE FOR : {}", thing);
+        // Get the new addressing lookup data
+        deviceLookupKey = getValidatedIdString();
+        initialize();
```

Comment:
I will cross check this and update this comment, I remember there being some behaviour it was there to work around.

Reply:
This was because, when its reconfigured in the base class it doesn't call initialize. If the config has changed it effectively need's to re-pull the meta-data about the device, and start-up polling if its a known device (from the data via the protocol). Rather than call the framework method's, I've remapped the code into a new function and called that from the config change, and the initialize override. In case initialize does other bit's this should remove any un-intended behaviours of the framework, I think.

Reply:
What do you mean by "reconfigured in the base class"? The framework calls `initialize()` on the Thing. If you need to do some operations in the sub classes, you can call these methods from with `initialize()`.

In your current code `configurationUpdated()` is not used.

Reply:
So I had a closer look your right - it looks like dead code from testing various scenario's after removing it. I've removed this configurationUpdated() handler now.